src/detect.h

```diff
@@ -127,6 +127,10 @@ enum DetectSigmatchListEnum {
 
     DETECT_SM_LIST_BASE64_DATA,
 
+    DETECT_SM_LIST_DNP3_DATA_MATCH,
+    DETECT_SM_LIST_DNP3_MATCH,
```

Comment:
Looks like this would involve converting the detection functions to AppLayerTxMatch variants. Is that what you had in mind?